proposals/002-Auttomaticly-reload-on-Kubernetes-config-change.md

```diff
@@ -0,0 +1,76 @@
+# Proposal 002 - Automatically trigger a deployment rollout of affected proxies on Kubernetes config change 
+
+<!-- TOC -->
+* [Proposal 002 - Automatically trigger a deployment rollout of affected proxies on Kubernetes config change](#proposal-002---automatically-trigger-a-deployment-rollout-of-affected-proxies-on-kubernetes-config-change-)
+* [Problem Statement](#problem-statement)
+* [Proposed Solution](#proposed-solution)
+  * [The details](#the-details)
+  * [Advantages](#advantages)
+  * [Limitations](#limitations)
+* [Rejected Alternatives](#rejected-alternatives)
+<!-- TOC -->
+# Problem Statement
+There are a wide array of Kubernetes resources which are used to create the configuration used by any given Kroxylicious proxy process. These resources are consumed and generated by multiple independent reconcilers (in Java Operator SDK terms). Which leads to a tangled web of resources, reconcilers and triggers. 
+
+We decided to split the operator into multiple reconcilers so that there was a single writer for each of our "intermediate" resources (KafkaService, VirtualKafkaCluster, etc), and thus it is much easier to reason about what changed and why. However, that means that we do not have a single global view of when a proxy's configuration needs to be re-generated.
+
+# Proposed Solution
+
+The Kroxylicious operator is built up from what we refer to as "leaf" (such as the Kafka Service Reconciler) and "aggregating" (with the Kafka Proxy Reconciler being the ultimate aggregator) reconcilers. 
```

Comment:
Feels like the `leaf` and `aggegator` terms could use a definition immediately. Also feels a bit funky having the `leaf` naming when VKC has children. `component reconciler` vs `aggregating reconciler`? Tough one.

Reply:
or `intermediate` as you've named it in the problem statement?

Reply:
I like aggegator and intermediate